### PR TITLE
ADAL Identity Core setup

### PR DESCRIPTION
**Pull request by RPangrle**
NOTE: This change does not include any functional code changes, but includes a ton of changes to build and project settings, as well as renaming of files. It's not meant to be a complete moving of files, merely a starting point to make sure everything is in a working state that others can build off of.

Here is the matching PR in ADAL: https://github.com/AzureAD/azure-activedirectory-library-for-objc/pull/1104

* Copy over some NSDictionary, NSString and NSURL utils from ADAL
* Changes to xcconfig files to ease them being included with ADAL xcconfigs

### CODE REVIEW

File: IdentityCore/src/MSIDDeviceId.h [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+#import <Foundation/Foundation.h>
+
+#include <sys/types.h>
+#include <sys/sysctl.h>
+#include <mach/machine.h>
+
+/*!
+    This class varies based on whether we're included in ADAL or MSAL, so the included library
+    must implement this class.
```
Comment (oldalton) [inline IdentityCore/src/MSIDDeviceId.h]: Just a general discussion:
DeviceId generation is actually almost the same for ADAL and MSAL, except of x-client-SKU (constant per library) and x-client-Ver. I was thinking of combining the generation code for them and have another SDK specific place for returning version (used in other places) and SDK name. I can do the change, just wanted to know what do you think?

File: IdentityCore/xcconfig/identitycore__common__ios.xcconfig [hunk 2]
```diff
@@ -1,6 +1,7 @@
-SDKROOT = iphoneos;
-CODE_SIGN_IDENTITY = "iPhone Developer";
-IPHONEOS_DEPLOYMENT_TARGET = 9.3;
-LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks @loader_path/Frameworks";
-TARGETED_DEVICE_FAMILY = 1,2;
-ENABLE_BITCODE = YES;
+SDKROOT = iphoneos
+CODE_SIGN_IDENTITY = "iPhone Developer"
+IPHONEOS_DEPLOYMENT_TARGET = 9.3
```
Comment (oldalton) [inline IdentityCore/xcconfig/identitycore__common__ios.xcconfig]: nit: should we have it 9.0 for clarity?

Comment (RPangrle) [reply IdentityCore/xcconfig/identitycore__common__ios.xcconfig]: There are API differences between 9.0 and 9.3, we only support the latest patched version on any major version of iOS.

File: IdentityCore/src/MSIDDeviceId.h (hunk 1 above)
Comment (RPangrle) [reply IdentityCore/src/MSIDDeviceId.h]: I'm trying to not make any functional code changes here, just copy and paste. This is such a large and confusing PR as is that my preference is for people to not have to think that there's code changes they need to be looking out for as well while reviewing.

**Pull request merged**